src/main/java/com/yahoo/sketches/pig/sampling/VarOptCommonImpl.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Copyright 2017, Yahoo! Inc.
+ * Licensed under the terms of the Apache License 2.0. See LICENSE file at the project root for terms.
+ */
+
+package com.yahoo.sketches.pig.sampling;
+
+import java.io.IOException;
+
+import org.apache.pig.EvalFunc;
+import org.apache.pig.backend.executionengine.ExecException;
+import org.apache.pig.data.BagFactory;
+import org.apache.pig.data.DataBag;
+import org.apache.pig.data.DataByteArray;
+import org.apache.pig.data.Tuple;
+import org.apache.pig.data.TupleFactory;
+
+import com.yahoo.memory.Memory;
+import com.yahoo.sketches.sampling.VarOptItemsSamples;
+import com.yahoo.sketches.sampling.VarOptItemsSketch;
+import com.yahoo.sketches.sampling.VarOptItemsUnion;
+
+/**
+ * A collection of methods and constants used across VarOpt UDFs.
+ *
+ * @author Jon Malkin
+ */
+class VarOptCommonImpl {
+  static final int DEFAULT_TARGET_K = 1024;
+  static final int DEFAULT_WEIGHT_IDX = 0;
+
+  static final String WEIGHT_ALIAS = "vo_weight";
+  static final String RECORD_ALIAS = "record";
+
+  private static final BagFactory BAG_FACTORY = BagFactory.getInstance();
+  private static final TupleFactory TUPLE_FACTORY = TupleFactory.getInstance();
+  private static final ArrayOfTuplesSerDe SERDE = new ArrayOfTuplesSerDe();
+
+  // Produces a sketch from a bag of input Tuples
+  static VarOptItemsSketch<Tuple> rawTuplesToSketch(final Tuple inputTuple,
+                                                    final int k, final int weightIdx)
+          throws IOException {
+    assert inputTuple != null;
+    assert inputTuple.size() >= 1;
+    assert !inputTuple.isNull(0);
+
+    final DataBag samples = (DataBag) inputTuple.get(0);
+    final VarOptItemsSketch<Tuple> sketch = VarOptItemsSketch.newInstance(k);
+
+    for (Tuple t : samples) {
+      // first element is weight
+      final double weight = (double) t.get(weightIdx);
+      sketch.update(t, weight);
+    }
+
+    return sketch;
+  }
+
+  // Produces a union from a bag of serialized sketches
+  static VarOptItemsUnion<Tuple> unionSketches(final Tuple inputTuple, final int k)
+          throws IOException {
+    assert inputTuple != null;
+    assert inputTuple.size() >= 1;
+    assert !inputTuple.isNull(0);
+
+    final VarOptItemsUnion<Tuple> union = VarOptItemsUnion.newInstance(k);
+
+    final DataBag sketchBag = (DataBag) inputTuple.get(0);
+    for (Tuple t : sketchBag) {
+      final DataByteArray dba = (DataByteArray) t.get(0);
+      final Memory mem = Memory.wrap(dba.get());
+      union.update(mem, SERDE);
+    }
+
+    return union;
+  }
+
+  // Serializes a sketch to a DataByteArray and wraps it in a Tuple
+  static Tuple wrapSketchInTuple(final VarOptItemsSketch<Tuple> sketch) throws IOException {
+    final DataByteArray dba = new DataByteArray(sketch.toByteArray(SERDE));
+    final Tuple outputTuple = TUPLE_FACTORY.newTuple(1);
+    outputTuple.set(0, dba);
```

Comment:
I believe I'll run into line length limits, but I can definitely shorten it to 2 lines